snc.sh

```diff
@@ -238,47 +238,43 @@ function create_pvs() {
 # in order to trigger regeneration of the initial 24h certs the installer created on the cluster
 function renew_certificates() {
+    # This random_string is created by installer and added to each resource type,
+    # in installer side also variable name is kept as `random_string`
+    # so to maintain consistancy, we are also using random_string here.
+    random_string=$(sudo virsh list --all | grep -oP "(?<=${CRC_VM_NAME}-).*(?=-master-0)")
+    if [ -z $random_string ]; then
+        echo "Could not find virtual machine created by snc.sh"
+        exit 1;
+    fi
+    VM_PREFIX=${CRC_VM_NAME}-${random_string}
+
+    sudo virsh shutdown ${VM_PREFIX}-master-0
+    # Wait till instance started successfully
+    until sudo virsh domstate ${VM_PREFIX}-master-0 | grep shut; do
+        echo " ${VM_PREFIX}-master-0 still running"
+        sleep 3
+    done
+
     # Enable the network time sync and set the clock back to present on host
     sudo date -s '1 day'
     sudo timedatectl set-ntp on
 
-    # Create the unsupported cert rotation config for a day (validity will increase by 30 times so it will become 30days)
-    ${OC} create -n openshift-config configmap unsupported-cert-rotation-config --from-literal='base=86400s'
-
-    # Delete the pods to have cert rotation
-    ${OC} delete pods --all -A --force --grace-period=0
-
-    # Wait till the api server again started
-     while ! ${OC} get nodes >/dev/null 2>&1; do
-         sleep 3
-     done
-     echo "API server is up"
-
-    # Force the rotation for secrets
-    ${OC} get secret -A -o json | ${JQ} -r '.items[] | select(.metadata.annotations."auth.openshift.io/certificate-not-after" | .!=null and fromdateiso8601<='$( date --date='+1year' +%s )') | "-n \(.metadata.namespace) \(.metadata.name)"' | \
-        xargs -n3 ${OC} patch secret -p='{"metadata": {"annotations": {"auth.openshift.io/certificate-not-after": null}}}'
+    sudo virsh start ${VM_PREFIX}-master-0
+    # Wait till ssh connection available
+    until ${SSH} core@api.${CRC_VM_NAME}.${BASE_DOMAIN} -- "exit 0" >/dev/null 2>&1; do
+        echo " ${VM_PREFIX}-master-0 still booting"
+        sleep 2
+    done
 
-    # Delete the current csr signer to get new request.
-    ${OC} delete secrets/csr-signer-signer secrets/csr-signer -n openshift-kube-controller-manager-operator || true
-    
-    echo "Waiting for 5 mins to make sure cluster is stable again"
-    sleep 300
+    # After cluster starts kube-apiserver-client-kubelet signer need to be approved
+    timeout 300 bash -c -- "until ${OC} get csr | grep Pending; do echo 'Waiting for first CSR request.'; sleep 2; done"
+    ${OC} get csr -ojsonpath='{.items[*].metadata.name}' | xargs ${OC} adm certificate approve
 
-    # Retry 4 times to make sure kubelet certs are rotated correctly.
+    # Retry 5 times to make sure kubelet certs are rotated correctly.
     i=0
-    while [ $i -lt 4 ]; do
+    while [ $i -lt 5 ]; do
```

Comment:
Seems fairly arbitrary? Wondering if we could do better? Maybe loop a lot more, but don't sleep 60s each time?

Reply:
I know, atm I don't know if we can have something better, I didn't change the 60s time but yes we can do it 30s and add more retry.